sdk/python/feast/feature_store.py

```diff
@@ -1929,13 +1922,24 @@ def retrieve_online_documents(
         document_feature_distance_vals = [feature[5] for feature in document_features]
         online_features_response = GetOnlineFeaturesResponse(results=[])
         requested_feature = requested_feature or requested_features[0]
+        if vector_field_metadata := _get_feature_view_vector_field_metadata(
+            requested_feature_view
+        ):
+            vector_field_name = vector_field_metadata.name
+        data = {
+            **join_key_values,
+            vector_field_name: document_feature_vals,
+            "distance": document_feature_distance_vals,
+        }
+        _requested_features = [_feature.split(":")[-1] for _feature in feature_list]
+        requested_features_data = {
+            _feature: data[_feature]
+            for _feature in _requested_features
+            if _feature in data
+        }
         utils._populate_result_rows_from_columnar(
             online_features_response=online_features_response,
-            data={
-                **join_key_values,
-                requested_feature: document_feature_vals,
-                "distance": document_feature_distance_vals,
```

Comment:
We expect the distance to be returned in this version of the API. I'd add it back . 👍

Reply:
the test will pass once you fix that.